yelp_loop.py

```diff
@@ -273,17 +273,39 @@ def get_location_from_azure(query):
 
     return longitude_west, longitude_east, latitude_south, latitude_north
 
-# Hack: using regex to change all location clauses
+def convert_opening_hours_query(opening_hours_query):
+    prompt = "You are a parser for turning natural language descriptions of time intervals in a week into the format [day].[starting_hour].[starting_minute].[ending_hour].[ending_minute].\n"
+    prompt += "The days of the week are 0-indexed with Monday = 0, Tuesday = 1 ... Saturday = 5, Sunday = 6, and the hour.minute representations are between 0.0 and 23.59.\n"
+    prompt += "Multiple intervals are demarcated with the hyphen '-', and if no particular day is specified, then you must generate the intervals for every day in the week.\n"
+    prompt += "Here is an example:\n USER_QUERY: after 2pm on Monday\n TRANSLATION: 0.14.0.23.59\n"
+    prompt += "Here is another example: \n USER_QUERY: between 7am and 2pm on Tuesday and Wednesday\n TRANSLATION: 1.7.0.14.0-2.7.0.14.0\n"
+    prompt += "Here is another example:\n USER_QUERY: before 4am\n TRANSLATION: 0.0.0.3.59-1.0.0.3.59-2.0.0.3.59-3.0.0.3.59-4.0.0.3.59-5.0.0.3.59-6.0.0.3.59\n"
+    prompt += "Here is another example:\n USER_QUERY: after 3pm\n TRANSLATION: 0.15.0.23.59-1.15.0.23.59-2.15.0.23.59-3.15.0.23.59-4.15.0.23.59-5.15.0.23.59-6.15.0.23.59\n"
+    prompt += "Here is an example:\n USER_QUERY: at 2pm on Monday\n TRANSLATION: 0.14.0.14.0\n"
+    prompt += "Here is an example:\n USER_QUERY: at 3am\n TRANSLATION: 0.3.0.3.0-1.3.0.3.0-2.3.0.3.0-3.3.0.3.0-4.3.0.3.0-5.3.0.3.0-6.3.0.3.0\n"
+    prompt += "Now it is your turn to generate the translation: USERY_QUERY " + opening_hours_query + "\n TRANSLATION: "
+    messages = [{"role":"system","content":prompt}]
+    response = openai.OpenAI().chat.completions.create(**{"messages":messages, "model":"gpt-3.5-turbo-0613"}).choices[0].message.content 
```

Comment:
Could you refactor to use the `llm_generate` function? Since everything else is calling that function.

Reply:
See new commit